src/docfx/restore/index/DependencySlotPool.cs

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.Docs.Build
+{
+    internal static class DependencySlotPool
+    {
+        private static int _defaultLockdownTimeInSecond = 10 * 60;
+
+        public static async Task<(string path, T slot)> TryGetSlot<T>(string remote, Func<IReadOnlyList<T>, IReadOnlyList<T>> matchExistingSlots) where T : DependencySlot
```

Comment:
possible to put `<T>` on class level to get rid of `where` constraint on every method?
if not, do we need where `T : new ()` here?

Reply:
we need`T :  new()` here to new a T.